### PR TITLE
overlord: add manager test for "assumes" checking

### PR DESCRIPTION
**Pull request by mvo5**
We want to check that `assumes` are handled correctly early. This
commit adds manager tests for `Install`, `Update` and `UpdateMany`
to ensure that snaps with the wrong `assumes` are rejected before
the taskset is generated.

This also required adding "snap-yaml" to the mockServer in
the manager tests.

This also includes https://github.com/snapcore/snapd/pull/10659

### CODE REVIEW

**Review by mardy — APPROVED** [review]
LGTM!

File: overlord/managers_test.go [hunk 1]
```diff
@@ -3272,6 +3283,97 @@ version: 1.0
...
+
+	mockServer := s.mockStore(c)
+	defer mockServer.Close()
+
+	st := s.o.State()
+	st.Lock()
+	defer st.Unlock()
+
+	_, err := snapstate.Install(context.TODO(), st, "some-snap", nil, 0, snapstate.Flags{})
+	c.Assert(err, ErrorMatches, `snap "some-snap" assumes unsupported features: something-that-is-not-provided \(try to update snapd and refresh the core snap\)`)
```
Comment (stolowski) [inline overlord/managers_test.go:3305]: Isn't the "..and refresh the core snap" part of the message going to be confusing when core18/20 is in use?

Comment (stolowski) [reply overlord/managers_test.go:3305]: Ah, sorry, I didn't notice this is just adding tests and not changing the code. Ignore for now.

**Review by stolowski — APPROVED** [review]
Nice, thank you

File: overlord/managers_test.go (hunk 1 above)
Comment (mvo5) [reply overlord/managers_test.go:3305]: Yes! I think it's something we want to fix too and just mention "base snap" or similar.

**Review by anonymouse64 — APPROVED** [review]
lgtm, thanks for this

**Review by pedronis — APPROVED** [review]
thanks for adding these tests

**Pull request merged**